price-feeder/oracle/provider/kraken.go

```diff
@@ -180,59 +208,64 @@ func (p *KrakenProvider) messageReceived(messageType int, bz []byte) {
 
 	var krakenEvent KrakenEvent
 	if err := json.Unmarshal(bz, &krakenEvent); err != nil {
-		p.logger.Debug().Msg("received a message that is not an event")
 		// msg is not an event, it will try to marshal to ticker message
-		p.messageReceivedTickerPrice(bz)
-		return
+		p.logger.Debug().Msg("received a message that is not an event")
+	} else {
+		switch krakenEvent.Event {
+		case krakenEventSystemStatus:
+			p.messageReceivedSystemStatus(bz)
+			return
+		case krakenEventSubscriptionStatus:
+			p.messageReceivedSubscriptionStatus(bz)
+			return
+		}
 	}
 
-	switch krakenEvent.Event {
-	case krakenEventSystemStatus:
-		p.messageReceivedSystemStatus(bz)
-		return
-	case krakenEventSubscriptionStatus:
-		p.messageReceivedSubscriptionStatus(bz)
+	if err := p.messageReceivedTickerPrice(bz); err != nil {
+		// msg is not a ticker, it will try to marshal to candle message
+		p.logger.Debug().Err(err).Msg("unable to unmarshal ticker")
+	} else {
 		return
 	}
+	if err := p.messageReceivedCandle(bz); err != nil {
```

Comment:
This will be executed if `messageReceivedTickerPrice` errors correct?

Reply:
Yup, logic is if `messageReceivedTickerPrice` errors, log and continue to attempt parsing as a candl; else, return and don't attempt to parse as a candle